### PR TITLE
feat: add db setup, migrations setup, dockerized postgres, update readme

### PR DESCRIPTION
**Pull request by eduardhyan**

### CODE REVIEW

File: migrations/20220424193304-create_users_table.sql [hunk 1]
```diff
@@ -0,0 +1,16 @@
+
+-- +migrate Up
+CREATE TABLE IF NOT EXISTS users
+(
+    id SERIAL,
+    username TEXT NOT NULL,
+    email TEXT,
+    password TEXT NOT NULL,
+    "dateCreated" DATE NOT NULL,
```
Comment (dominikvrbic) [inline migrations/20220424193304-create_users_table.sql]: not sure is this an example or what but i would setup a
 created_at TIMESTAMPTZ 
 updated_at TIMESTAMPTZ 
that can be reused for all tables (i can do it in a pr later)

**Comment by dominikvrbic** [comment]
rest looks fine just add the end of file new lines where github is marking there should be @FrostyJet

File: migrations/20220424193304-create_users_table.sql (hunk 1 above)
Comment (eduardhyan) [reply migrations/20220424193304-create_users_table.sql]: what is the benefits of that approach rather than this one?
There laravel style vs nodejs :D

Comment (dominikvrbic) [reply migrations/20220424193304-create_users_table.sql]: i just mean putting TIMESTAMPTZ not DATE to get all of the data cos. we will need it for the timelines i guess, and making pg take care of it

Comment (eduardhyan) [reply migrations/20220424193304-create_users_table.sql]: Oh right, didn't notice that part. You're right, will update now, also just noticed that I used camel-case, will change to snake-case

Comment (dominikvrbic) [reply migrations/20220424193304-create_users_table.sql]: in that case do this as well https://x-team.com/blog/automatic-timestamps-with-postgresql/

File: migrations/20220424193304-create_users_table.sql [hunk 2]
```diff
@@ -0,0 +1,16 @@
+
+-- +migrate Up
+CREATE TABLE IF NOT EXISTS users
+(
+    id SERIAL,
```
Comment (dominikvrbic) [inline migrations/20220424193304-create_users_table.sql]: also any reason why not just to do ` id SERIAL NOT NULL PRIMARY KEY`

File: migrations/20220424193304-create_users_table.sql (hunk 1 above)
Comment (eduardhyan) [reply migrations/20220424193304-create_users_table.sql]: that is out of this PR's scope sorry

Comment (dominikvrbic) [reply migrations/20220424193304-create_users_table.sql]: will make a task for it then

File: migrations/20220424193304-create_users_table.sql (hunk 2 above)
Comment (eduardhyan) [reply migrations/20220424193304-create_users_table.sql]: done

Comment (dominikvrbic) [reply migrations/20220424193304-create_users_table.sql]: then you dont need this at all `CONSTRAINT users_pkey PRIMARY KEY (id)`

File: migrations/20220424193304-create_users_table.sql (hunk 1 above)
Comment (eduardhyan) [reply migrations/20220424193304-create_users_table.sql]: awesome, thanks

**Review by dominikvrbic — APPROVED** [review]
looks good

**Pull request merged**